eng/generators.targets

```diff
@@ -1,51 +1,56 @@
     <EnableLibraryImportGenerator Condition="'$(EnableLibraryImportGenerator)' == '' and
                                              '$(MSBuildProjectName)' == 'System.Private.CoreLib'">true</EnableLibraryImportGenerator>
   </PropertyGroup>
 
   <ItemGroup>
     <EnabledGenerators Include="LibraryImportGenerator" Condition="'$(EnableLibraryImportGenerator)' == 'true'" />
-    <!-- If the current project is not System.Private.CoreLib, we enable the LibraryImportGenerator source generator
+    <!-- We enable the Downlevel LibraryImportGenerator
          when the project is a C# source or test project that:
-         - doesn't target the a TFM that includes LibraryImportGenerator or
-         - doesn't reference the live targeting pack (i.e. when inbox) and
-           - references System.Private.CoreLib, or
-           - references System.Runtime.InteropServices -->
-    <EnabledGenerators Include="LibraryImportGenerator"
+         - doesn't target the a TFM that includes LibraryImportGenerator -->
+    <EnabledGenerators Include="DownlevelLibraryImportGenerator"
                        Condition="'$(EnableLibraryImportGenerator)' == '' and
                                   (
                                     '$(IsSourceProject)' == 'true' or
                                     '$(IsTestProject)' == 'true' or
                                     '$(IsTestSupportProject)' == 'true'
                                   ) and
                                   '$(MSBuildProjectExtension)' == '.csproj' and
+                                  !$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', '$(NetCoreAppMinimum)'))" />
+
+    <!-- We enable the LibraryImportGenerator source generator
+         when the project is a C# source project that:
+         - references System.Private.CoreLib directly
+         - references System.Runtime.InteropServices directly and not through the live targeting pack (i.e. when inbox) -->
+    <EnabledGenerators Include="LibraryImportGenerator"
+                       Condition="'$(EnableLibraryImportGenerator)' == '' and
+                                  '$(IsSourceProject)' == 'true' and
+                                  '$(MSBuildProjectExtension)' == '.csproj' and
                                   (
-                                    !$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net7.0')) or
-                                    (
                                       '$(DisableImplicitFrameworkReferences)' == 'true' and
                                       (
                                         '@(Reference->AnyHaveMetadataValue('Identity', 'System.Runtime.InteropServices'))' == 'true' or
-                                        '@(ProjectReference->AnyHaveMetadataValue('Identity', '$(CoreLibProject)'))' == 'true'
+                                        '@(Reference->AnyHaveMetadataValue('Identity', 'System.Private.CoreLib'))' == 'true'
```

Comment:
This change looks wrong. System.Private.CoreLib is never a reference as it isn't part of the targeting pack.